File: common/src/main/java/com/pluxity/facility/FacilityRepository.java
package com.pluxity.facility;

import java.util.List;
import java.util.Optional;
import lombok.NonNull;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface FacilityRepository extends JpaRepository<Facility, Long> {

    @NonNull
    List<Facility> findAll();

    @NonNull
    Optional<Facility> findById(@NonNull Long id);

    boolean existsByCode(String code);

    Optional<Facility> findByCode(String code);
}

File: common/src/main/java/com/pluxity/facility/FacilityRepository.kt
package com.pluxity.facility

import com.pluxity.global.annotation.CheckPermission
import com.pluxity.user.entity.ExecutionPhase
import com.pluxity.user.entity.PermissionType
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface FacilityRepository : JpaRepository<Facility, Long> {
    fun existsByCode(code: String): Boolean

    fun findByCode(code: String): Facility?

    @CheckPermission(type = PermissionType.ID, phase = ExecutionPhase.FILTER)
    fun findAllByOrderByCreatedAtDesc(): List<Facility>
}
